### PR TITLE
 android-interop-testing/examples: upgrade android plugin

### LINKED ISSUE #5523: Android Interopt Test is breaking due to mockito upgrade
See https://github.com/grpc/grpc-java/pull/5484#issuecomment-478637238

cc @TimvdLippe

### PR DESCRIPTION
**Pull request by dapengzhang0**
This resolves #5523

While bumping `com.android.tools.build:gradle:3.1.2` to `3.3.0`, some other plugins/artifacts/maven repo/buildscripts have to be updated:

- gradle (wrapper) need to upgrade to 4.10.x
- protobuf gradle plugin need to bump a version compatible with gradle version.
- need add `google()` and `jcenter()` repos for android (otherwise `com.android.tools.build:aapt2:3.3.0x` and `trove4j` will not be found resp.)
- need to accept license for Android "build-tools;28.0.3" in kokoro env.

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -36,6 +36,8 @@ android {
 }
 
 repositories {
+    google()
```
Comment (ejona86) [inline android/build.gradle]: Why were these added?

Comment (dapengzhang0) [reply android/build.gradle]: I had a build error "Could not find com.android.tools.build:aapt2:3.3.0-5013011".

File: buildscripts/kokoro/android-interop.sh [hunk 2]
```diff
@@ -15,6 +15,9 @@ export LDFLAGS=-L/tmp/protobuf/lib
 export CXXFLAGS=-I/tmp/protobuf/include
 export LD_LIBRARY_PATH=/tmp/protobuf/lib
 export OS_NAME=$(uname)
+export ANDROID_SDK_HOME=/opt/android-sdk/current
```
Comment (ejona86) [inline buildscripts/kokoro/android-interop.sh]: ANDROID_HOME is already defined with this value.

File: buildscripts/kokoro/android-interop.sh [hunk 3]
```diff
@@ -15,6 +15,9 @@ export LDFLAGS=-L/tmp/protobuf/lib
...
+
+echo y | ${ANDROID_SDK_HOME}/tools/bin/sdkmanager "build-tools;28.0.3"
```
Comment (ejona86) [inline buildscripts/kokoro/android-interop.sh]: Hmm... this will actually install this particular version. I sort of question if it would be better to drop the right MD5s into licenses/android-sdk-license

File: buildscripts/kokoro/android-interop.sh (hunk 2 above)
Comment (dapengzhang0) [reply buildscripts/kokoro/android-interop.sh]: Is `ANDROID_HOME` already defined for both `android` and `android-interop-test` in kokoru env?

Comment (ejona86) [reply buildscripts/kokoro/android-interop.sh]: Kokoro defines ANDROID_HOME. I think the android gradle plugin itself requires ANDROID_HOME to be set.

File: buildscripts/kokoro/android-interop.sh (hunk 3 above)
Comment (dapengzhang0) [reply buildscripts/kokoro/android-interop.sh]: linux-artifacts is also failing. The same issue. http://g/kokoro-users/_P4EAOWi-ko/PRFQvowPDgAJ

**Comment by dapengzhang0** [comment]
Now Android Interop is passing in kokoro.

File: android/build.gradle (hunk 1 above)
Comment (ejona86) [reply android/build.gradle]: That explains `google`, but why jcenter?

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.9-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.1-all.zip
```
Comment (ejona86) [inline gradle/wrapper/gradle-wrapper.properties]: -bin (to match the others, and because it is smaller)

File: android/build.gradle (hunk 1 above)
Comment (dapengzhang0) [reply android/build.gradle]: After fixing  "Could not find com.android.tools.build:aapt2:3.3.0-5013011" by adding `google()`, there is another artifact missing `Could not find org.jetbrains.trove4j:trove4j`.

File: buildscripts/kokoro/android-interop.sh (hunk 3 above)
Comment (dapengzhang0) [reply buildscripts/kokoro/android-interop.sh]: >I sort of question if it would be better to drop the right MD5s into licenses/android-sdk-license

I don't understand how to do that/ how that works.

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (dapengzhang0) [reply gradle/wrapper/gradle-wrapper.properties]: Done.

File: buildscripts/kokoro/android-interop.sh (hunk 3 above)
Comment (ejona86) [reply buildscripts/kokoro/android-interop.sh]: Hmm... it seems the license handling is done differently these days. Previously after you accepted a licenses you could get any of the binaries with that license. But that doesn't seem to work any more. So this seems fine.
